service/matching/taskListManager.go

```diff
@@ -741,7 +727,37 @@ func (c *taskListManagerImpl) isTaskAddedRecently(lastAddTime time.Time) bool {
 	return time.Now().Sub(lastAddTime) <= c.config.MaxTasklistIdleTime()
 }
 
-// if domainCache return error, it will return "" as domainNamne and a scope without domainName tagged
+func (c *taskListManagerImpl) domainScope() metrics.Scope {
+	scope := c.domainScopeValue.Load().(metrics.Scope)
+	if scope != nil {
+		return scope
+	}
+	c.tryInitDomainNameAndScope()
+	return c.domainScopeValue.Load().(metrics.Scope)
```

Comment:
what does this return if the init domain scope does not happen?

Reply:
constructor always initializes to a valid value. If domain lookup fails, the atomic is initialized with ordinary scope (without domain tag). tryInit() will always transition from one valid state to another.. so, to answer your question, Load() will never return nil